### PR TITLE
fix: fixed the mac binary release tag pattern

### PR DESCRIPTION
**Pull request by wacban**
tldr - regexes are hard

The mac_binary.yml configures that macbook binaries should be uploaded to S3 for any commit pushed to master as well as any tag matching the given pattern. Unforunately the pattern is invalid and it doesn't capture our release tags and so we never upload mac binaries. This breaks a few things including `nearup localnet` and `db_migration.py` pytest.

I have no clue how to test this PR so I'm looking for suggestions. It's broken anyway so I don't think I'll make it any worse though so merging without a test would be acceptable too. 

references:
- [6426](https://github.com/near/nearcore/pull/6426/files#r831461185)  the PR that introduced the tag trigger and discussion about whether the right pattern is `.*` or `*` 
- [mac_binary](https://github.com/near/nearcore/actions/workflows/mac_binary.yml) Is the github action that is triggered by the mac_binary.yml. You can see that it only gets triggered for commits pushed to master but there aren't any runs release tags. 
- [workflow-syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) Shows that indeed the pattern for capturing any string (excl. `/`) is `*`.

### CODE REVIEW

**Review by Ekleog-NEAR — APPROVED** [review]

**Comment by nagisa** [comment]
The best way I found to test github actions is to either push straight into the primary repository, or set up a copy of the repo (not a fork), and test them there. There aren’t any local ways to test actions as far as I can tell.

**Comment by nagisa** [comment]
Question: are we sure we don’t want to also deploy the `-rc` mac binaries?

**Review by nagisa — APPROVED** [review]

**Comment by wacban** [comment]
> Question: are we sure we don’t want to also deploy the `-rc` mac binaries?

I believe that the pattern that I'm proposing here - '[0-9]+.[0-9]+.[0-9]+*' - should include the -rc tags too 
(e.g. 1.31.0-rc.4 should be fine).

File: .github/workflows/mac_binary.yml [hunk 1]
```diff
@@ -4,7 +4,8 @@ on:
     branches:
       - master
     tags:
-      - '[0-9]+.[0-9]+.[0-9]+.*'
+      - '[0-9]+.[0-9]+.[0-9]+*'
+    workflow_dispatch:
```
Comment (nagisa) [inline .github/workflows/mac_binary.yml:8]:
```suggestion
  workflow_dispatch:
```

I believe, this needs to be `on.workflow_dispatch` not `on.push.workflow_dispatch`.

Comment (nagisa) [reply .github/workflows/mac_binary.yml:8]: Ideally this is put in between `on` and `push` so it is easier to keep track of identation.

**Review by nagisa — APPROVED** [review]
